controlplane/kubeadm/internal/workload_cluster_etcd.go

```diff
@@ -85,6 +84,10 @@ func (w *Workload) EtcdIsHealthy(ctx context.Context) (HealthCheckResult, error)
 		}
 		defer etcdClient.Close()
 
+		if err := etcdClient.HealthCheck(ctx); err != nil {
```

Comment:
Add a comment above describing what this health check does?

Reply:
+1